resources/js/entities/User.js

```diff
@@ -73,14 +73,14 @@ export default class User {
   }
 
   setPassword(newPw, oldPw) {
-    if (typeof newPw === "string" && typeof oldPw === "string") {
-      if (oldPw === this._password) {
-        this._password = newPw;
-      } else {
-        throw new Error("Old password is incorrect.");
-      }
-    } else {
+    if (typeof newPw !== "string" || typeof oldPw !== "string") {
       throw new TypeError("Expected two strings.");
     }
+
+    if (oldPw !== newPw) {
```

Comment:
El siguiente if sirve para validar si el pasword actual 'this._password' es igual al password ingresado por el usuario (OldPw) ?

Reply:
sí, me equivoque cuando escribí la varibiable
era `(oldPw === this._password)`